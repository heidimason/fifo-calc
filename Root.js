import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { combineReducers } from 'redux'

import PurchaseHistoryReducer from './src/reducers/PurchaseHistory'
import SaleHistoryReducer from './src/reducers/SaleHistory'

const rootReducer = combineReducers({
		purchaseHistory: PurchaseHistoryReducer,
		saleHistory: SaleHistoryReducer
	}),

	store = createStore(
		rootReducer
	)

export default props => {
	return (
		<Provider store={store}>
			{props.children}
		</Provider>
	)
}
