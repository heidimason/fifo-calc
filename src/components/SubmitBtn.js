import React from 'react'
import { Text, TouchableOpacity } from 'react-native'
import { btns } from '../utils/styles/btns'
import { fonts } from '../utils/styles/fonts'

const SubmitBtn = ({ children, disabled, onPress, style = {} }) => {
  	return (
    	<TouchableOpacity
    		disabled={disabled}
        	onPress={onPress}
        	style={style}>
        	<Text
                style={[btns.btnText, fonts.h2]}>{children}
            </Text>
    	</TouchableOpacity>
  	)
}

export default SubmitBtn