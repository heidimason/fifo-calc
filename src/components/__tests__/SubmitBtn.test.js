import React from 'react'
import renderer from 'react-test-renderer'

import Root from '../../../Root'
import SubmitBtn from '../SubmitBtn'

describe('<SubmitBtn />', () => {
	const tree = renderer.create(
		<Root>
			<SubmitBtn />
		</Root>
	).toJSON()

	it('renders correctly', () => {
		expect(tree).toMatchSnapshot()
	})

	it('has 1 child', () => {
		expect(tree.children.length).toBe(1)
	})
})
