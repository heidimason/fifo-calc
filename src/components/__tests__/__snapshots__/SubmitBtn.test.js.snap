// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitBtn /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontWeight": "bold",
          "textAlign": "center",
        },
        Object {
          "fontSize": 40,
        },
      ]
    }
  />
</View>
`;
